api/types/network/network.go

```diff
@@ -19,6 +19,30 @@ const (
 	NetworkNat = "nat"
 )
 
+// CreateRequest is the request message sent to the server for network create call.
+type CreateRequest struct {
+	CreateOptions
+	Name string // Name is the requested name of the network.
+}
+
+// CreateOptions holds options to create a network.
+type CreateOptions struct {
+	// Deprecated: CheckDuplicate is deprecated since API v1.44, but it defaults to true when sent by the client
+	// package to older daemons.
+	CheckDuplicate bool              `json:",omitempty"`
+	Driver         string            // Driver is the driver-name used to create the network (e.g. `bridge`, `overlay`)
```

Comment:
Actually let me push one small formatting change that will reduce the diff for my follow-up;

Reply:
done; this is the diff I pushed; whitespace changes only :smile:

```patch
diff --git a/api/types/network/network.go b/api/types/network/network.go
index f8be85d456..5992c3ace3 100644
--- a/api/types/network/network.go
+++ b/api/types/network/network.go
@@ -29,18 +29,19 @@ type CreateRequest struct {
 type CreateOptions struct {
        // Deprecated: CheckDuplicate is deprecated since API v1.44, but it defaults to true when sent by the client
        // package to older daemons.
-       CheckDuplicate bool              `json:",omitempty"`
-       Driver         string            // Driver is the driver-name used to create the network (e.g. `bridge`, `overlay`)
-       Scope          string            // Scope describes the level at which the network exists (e.g. `swarm` for cluster-wide or `local` for machine level).
-       EnableIPv6     *bool             `json:",omitempty"` // EnableIPv6 represents whether to enable IPv6.
-       IPAM           *IPAM             // IPAM is the network's IP Address Management.
-       Internal       bool              // Internal represents if the network is used internal only.
-       Attachable     bool              // Attachable represents if the global scope is manually attachable by regular containers from workers in swarm mode.
-       Ingress        bool              // Ingress indicates the network is providing the routing-mesh for the swarm cluster.
-       ConfigOnly     bool              // ConfigOnly creates a config-only network. Config-only networks are place-holder networks for network configurations to be used by other networks. ConfigOnly networks cannot be used directly to run containers or services.
-       ConfigFrom     *ConfigReference  // ConfigFrom specifies the source which will provide the configuration for this network. The specified network must be a config-only network; see [CreateOptions.ConfigOnly].
-       Options        map[string]string // Options specifies the network-specific options to use for when creating the network.
-       Labels         map[string]string // Labels holds metadata specific to the network being created.
+       CheckDuplicate bool `json:",omitempty"`
+
+       Driver     string            // Driver is the driver-name used to create the network (e.g. `bridge`, `overlay`)
+       Scope      string            // Scope describes the level at which the network exists (e.g. `swarm` for cluster-wide or `local` for machine level).
+       EnableIPv6 *bool             `json:",omitempty"` // EnableIPv6 represents whether to enable IPv6.
+       IPAM       *IPAM             // IPAM is the network's IP Address Management.
+       Internal   bool              // Internal represents if the network is used internal only.
+       Attachable bool              // Attachable represents if the global scope is manually attachable by regular containers from workers in swarm mode.
+       Ingress    bool              // Ingress indicates the network is providing the routing-mesh for the swarm cluster.
+       ConfigOnly bool              // ConfigOnly creates a config-only network. Config-only networks are place-holder networks for network configurations to be used by other networks. ConfigOnly networks cannot be used directly to run containers or services.
+       ConfigFrom *ConfigReference  // ConfigFrom specifies the source which will provide the configuration for this network. The specified network must be a config-only network; see [CreateOptions.ConfigOnly].
+       Options    map[string]string // Options specifies the network-specific options to use for when creating the network.
+       Labels     map[string]string // Labels holds metadata specific to the network being created.
 }

 // ListOptions holds parameters to filter the list of networks with.
```